### PR TITLE
Use UV instead of PDM

### PR DESCRIPTION
**Pull request by Kludex**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -139,20 +119,37 @@ testing = [
...
-    "mypy",
+    # linting
+    "eval-type-backport>=0.1.3",
+    "ruff==0.5.6",
+    # typing
+    "mypy~=1.1.1",
     "pydantic-settings>=2.0.0",
 ]
+
+[tool.pdm.dev-dependencies]
```
Comment (sydney-runkle) [inline pyproject.toml:137]: Why do we still have `pdm` stuff here?

File: pyproject.toml [hunk 2]
```diff
@@ -139,20 +119,37 @@ testing = [
     "pytest-codspeed~=2.2.0; python_version < '3.13'",
     "packaging>=23.2",
     "jsonschema>=4.23.0",
-]
-testing-extra = [
```
Comment (sydney-runkle) [inline pyproject.toml:143]: Hmm, don't we still want to group these?

Comment (gotmax23) [reply pyproject.toml:143]: https://github.com/astral-sh/uv/issues/5632

**Comment by Kludex** [comment]
PR welcome to do this, whoever wants to.

**Pull request closed without merging**